after_success.sh

```diff
@@ -19,8 +19,34 @@ else
         start_section "package.upload" "${GREEN}Package uploading...${NC}"
         # Test `anaconda` with ANACONDA_TOKEN before uploading
         source $GITHUB_WORKSPACE/.github/scripts/test_anaconda.sh
+
+        os_package_match='conda-bld/(.*)'
+        [[ $CONDA_OUT =~ $os_package_match ]]
+        os_and_package="${BASH_REMATCH[1]}"
+
+        name_version_match='conda-bld/(.*)/(.*)-(.*)-'
+        [[ $CONDA_OUT =~ $name_version_match ]]
+        name="${BASH_REMATCH[2]}"
+        version="${BASH_REMATCH[3]}"
+
+        check_path="$ANACONDA_USER/$name/$version/$os_and_package"
+        # This command may return non-zero exit code when
+        # the package doesn't exist in this version in Anaconda
+        # and it's okay
+        set +e
+        labels=$(anaconda show $check_path |& egrep "^labels.*'main'")
```

Comment:
I think we can simply check the return code here, give a message and call false in case the return code is `0`. If we don't use the `--force` switch and there's the same package on Anaconda (it definitely is if the return code is `0`) then the `anaconda upload` will fail.

The `--force` is used only in case there is the same package on any label apart from `main` so it can be always used.

Reply:
We can't check the return code if "set -e" is turned on at the beginning of this script. Script will exit immediately after "anaconda show" command.

Reply:
My comment wasn't about setting or unsetting the `e` bashopt. I meant that if the package is in `main` then we shouldn't even call `anaconda upload` (without `--force` it will always fail so there's no point to do that). This simplifies the further script.

It's true that you can't check the return code after calling the failing command (unless you do `|| true` at the end) but you certainly can do it like:
```bash
if anaconda show $check_path |& egrep "^labels.*'main'"; then
    echo "Package $check_path is present in 'main' label. Uploading will be skipped because doing so would remove the 'main' one."
    exit -1
else
    echo "Package $check_path not present in 'main' label"
    anaconda -t $ANACONDA_TOKEN upload --force --no-progress --user $ANACONDA_USER --label ci-$branch-$GITHUB_RUN_ID $CONDA_OUT
fi
```

Reply:
Applied solution proposed by @ajelinski